core/chains/evm/client/rpc_client.go

```diff
@@ -456,7 +456,12 @@ func (r *RPCClient) SubscribeToHeads(ctx context.Context) (ch <-chan *evmtypes.H
 	if r.newHeadsPollInterval > 0 {
 		interval := r.newHeadsPollInterval
 		timeout := interval
-		poller, channel := commonclient.NewPoller[*evmtypes.Head](interval, r.latestBlock, timeout, r.rpcLog)
+		poller, channel := commonclient.NewPoller[*evmtypes.Head](interval, func(pollRequestCtx context.Context) (*evmtypes.Head, error) {
+			if commonclient.CtxIsHeathCheckRequest(ctx) {
+				pollRequestCtx = commonclient.CtxAddHealthCheckFlag(pollRequestCtx)
+			}
+			return r.latestBlock(pollRequestCtx)
```

Comment:
Juggling multiple contexts in the same scope with different names is typically a code smell. Why don't we check the outer `ctx` just once before?
```suggestion
		isHealthCheckRequest := commonclient.CtxIsHeathCheckRequest(ctx)
		poller, channel := commonclient.NewPoller[*evmtypes.Head](interval, func(ctx context.Context) (*evmtypes.Head, error) {
			if isHealthCheckRequest {
				ctx = commonclient.CtxAddHealthCheckFlag(ctx)
			}
			return r.latestBlock(pollRequestCtx)
```
Also, out-of-scope, but these kinds of methods are typically called `With` rather than `Add` since contexts are not mutable.